File: Thales/musicplayer/src/main/java/com/example/musicplayer/playbackservice/MusicPlayerBroadcastLisener.java
package com.example.musicplayer.playbackservice;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.example.musicplayer.player.MusicPlayer;

public class MusicPlayerBroadcastLisener extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        //move to DI
        MusicPlayer musicPlayer = new MusicPlayer(context);
        switch (intent.getAction()) {
            case "com.example.musicplayer.ACTION_PLAY":
                musicPlayer.resume();
                break;
            case "com.example.musicplayer.ACTION_PAUSE":
                musicPlayer.pause();
                break;
            case "com.example.musicplayer.ACTION_STOP":
                musicPlayer.stop();
                break;
            case "com.example.musicplayer.ACTION_CLOSE":
                musicPlayer.stop();
                musicPlayer.clear();
                break;
            case "com.example.musicplayer.ACTION_PLAYLIST":
                Intent i = new Intent("your.custom.ACTION");
                i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                context.startActivity(i);
                break;
        }
    }
}

File: Thales/musicplayer/src/main/java/com/example/musicplayer/playbackservice/MusicPlayerBroadcastLisener.kt
package com.example.musicplayer.playbackservice

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

import com.example.musicplayer.player.MusicPlayer

class MusicPlayerBroadcastLisener : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {

        val musicPlayer = MusicPlayer(context)
        when (intent.action) {
            "com.example.musicplayer.ACTION_PLAY" -> musicPlayer.resume()
            "com.example.musicplayer.ACTION_PAUSE" -> musicPlayer.pause()
            "com.example.musicplayer.ACTION_STOP" -> musicPlayer.stop()
            "com.example.musicplayer.ACTION_CLOSE" -> {
                musicPlayer.stop()
                musicPlayer.clear()
            }
            "com.example.musicplayer.ACTION_PLAYLIST" -> {
                val i = Intent("your.custom.ACTION")
                i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
                context.startActivity(i)
            }
        }
    }
}